configuration/configuration.go

```diff
@@ -123,6 +127,46 @@ type Configuration struct {
 	} `yaml:"redis,omitempty"`
 }
 
+// LogHook is composed of hook Level and Type.
+// After hooks configuration, it can execute the next handling automatically,
+// when defined levels of log message emitted.
+// Example: hook can sending an email notification when error log happens in app.
+type LogHook struct {
+	// Type allows user to select which type of hook handler they want.
+	Type string `yaml:"type,omitempty"`
+
+	// Levels set which levels of log message will let hook executed.
+	Level []string `yaml:"levels,omitempty"`
+
+	// Mail allows user to configurate email parameters.
+	Mail Mail `yaml:"email,omitempty"`
```

Comment:
Make the yaml name consistent with the struct name. Just use `mail`, since we know we aren't referring to snail mail.